File: 035.SearchInsertPosition/MarkCSIE/SearchInsertPosition/src/SearchInsertPosition.java

public class SearchInsertPosition {

    public static void main(String[] args) {
        SearchInsertPosition test = new SearchInsertPosition();
        int[] testCase1 = new int[]{1, 3, 5, 6};
        System.out.println(test.searchInsert(testCase1, 5));
    }
    
    // [1,3,5,6]
    public int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        while (left <= right) {
            int middle = (left + right) / 2;
            if (nums[middle] < target) {
                left = middle + 1;
            } else {
                right = middle - 1;
            }
        }
        return left;
    }
}

File: 035.SearchInsertPosition/wyrj/SearchInsertPosition.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    var left = 0;
    var right = nums.length - 1;
    var mid;
    while (left <= right) {
        mid = Math.floor((left + right) / 2);
        if (nums.length - 1 === mid && nums[mid] < target) {
            return mid + 1;
        }
        if (nums[mid] >= target && (0 === mid || target > nums[mid - 1])) {
            return mid;
        }
        if (nums[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
};